### PR TITLE
Several smaller fixes, particularly in handling errors.

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

File: common/identity_parser/identity_parser.cpp [hunk 1]
```diff
@@ -29,7 +29,8 @@ kvp::ExpectedKeyValuesMap GetIdentityData(const string &identity_data_generator)
 	procs::Process proc({identity_data_generator});
 	auto ex_line_data = proc.GenerateLineData();
 	if (!ex_line_data) {
-		return expected::unexpected(ex_line_data.error());
+		return expected::unexpected(
+			ex_line_data.error().WithContext("While getting identity data"));
```
Comment (oleorhagen) [inline common/identity_parser/identity_parser.cpp:33]: We should have more of these :see_no_evil: :eyes:

File: mender-auth/cli/cli_test.cpp [hunk 2]
```diff
@@ -133,12 +136,24 @@ TEST(CliTest, DoAuthenticationCycleOnBootstrap) {
...
 
+	{
+		ASSERT_EQ(path::CreateDirectory(path::Join(tmpdir.Path(), "identity")), error::NoError);
+		auto path = path::Join(tmpdir.Path(), "identity", "mender-device-identity");
+		ofstream fd(path);
+		fd << R"(#!/bin/sh
+echo ID=ABC
+)";
+		ASSERT_TRUE(fd.good());
+		fs::permissions(path, fs::perms::owner_all);
```
Comment (oleorhagen) [inline mender-auth/cli/cli_test.cpp:147]: Nit: we have a `permissions` function in common don't we? :thinking:

**Review by oleorhagen — APPROVED** [review]

File: mender-update/update_module/v3/platform/c++17/fs_operations.cpp [hunk 3]
```diff
@@ -168,7 +156,10 @@ error::Error UpdateModule::CleanAndPrepareFileTree(
 	//
 
 	const fs::path header_subdir_path = file_tree_path / "header";
-	CreateDirectories(header_subdir_path);
+	err = path::CreateDirectories(header_subdir_path);
```
Comment (lluiscampos) [inline mender-update/update_module/v3/platform/c++17/fs_operations.cpp:159]:
```suggestion
	err = path::CreateDirectories(header_subdir_path.string());
```

**Review by lluiscampos — APPROVED** [review]
Good work, thanks for going through so many `filesystem` exception-throwing functions

**Pull request merged**